docs/developers/testing.md

```diff
@@ -0,0 +1,314 @@
+```
+/var/lib/irods/unit_tests/TEST_NAME
+```
+The Catch2 tests are built as executable files - one for each test suite - that can only be executed one at a time. Each executable comes with a runtime options provided by Catch2.
+
+### How to add tests
+
+There are three parts to every Catch2 test suite: a C++ source file, a CMake target, and an entry in the unit tests list.
+
+#### Add a new test
+
+The unit tests are maintained in the main iRODS repository. The test files are located in the `unit_tests/src` directory.
+
+Adding a new test file or test case is a matter of discretion on the part of the developer. Here are some hints for helping to make the decision:
+
+ 1. If you are able to locate an existing file which seems related to the feature or issue being tested but your use case requires a specific kind of setup, add a new test case to the existing test file.
+ 1. If you are able to locate an existing test case which seems related to the feature or issue being tested, add a new section to the existing test case.
+ 3. If you are unable to find an existing file which seems related to the feature or issue being tested, or you wish to keep your tests separated, add a new test file with its own test cases.
+ 4. If you are able to locate an existing test case which is precisely related to the feature or issue being tested, modify the existing test case to cover the new functionality.
+
+You will almost always want to add at least a new section to an existing test case, if not a new test case or even a new test file. Modify an existing test case (option 4) should only happen if you are fixing an existing test which is broken, or requirements changed in such a way that existing test cases must be adjusted.
+
+If you add a new test file, place it in the `unit_tests/src` directory with a name beginning with `test_`. For example, if you are testing a new library called `my_cool_library`, the unit test file could be called `test_my_cool_library.cpp`.
+
+See the [Catch2 documentation](https://github.com/catchorg/Catch2) for information on writing Catch2 tests.
+
+#### Add a CMake target
+
+If you add a new test file, a CMake file will need to be added to the `unit_tests/cmake/test_config` directory and the `subdirectory` added to the main `unit_tests` CMakeLists.txt. The CMake file should have a name that begins with `irods_`. It should be named in such a way that it can be connceted to the source file with which it is associated. Continuing with the previous example, a CMake file for testing `my_cool_library` should be named `irods_my_cool_library.cmake`.
```

Comment:
```suggestion
If you add a new test file, a CMake file will need to be added to the `unit_tests/cmake/test_config` directory and the `subdirectory` added to the main `unit_tests` CMakeLists.txt. The CMake file should have a name that begins with `irods_`. It should be named in such a way that it can be connected to the source file with which it is associated. Continuing with the previous example, a CMake file for testing `my_cool_library` should be named `irods_my_cool_library.cmake`.
```